doc/decisions/map_undo_redo_implementation.md

```diff
@@ -0,0 +1,146 @@
+# Map Undo/Redo Implementation
+
+## Problem
+
+Undo/redo functionality is a common feature in many applications, PermaplanT is no exception.
+The user should be able to undo/redo the actions taken within the canvas.
+
+## Constraints
+
+1. The implementation of undo/redo should not have a significant impact on the performance of the app.
+
+## Assumptions
+
+N/A
+
+## Solutions
+
+### Store every step of the user in the database
+
+The idea is to store every step of the user taken within the canvas in the database.
+The user can then undo/redo the actions by fetching the previous/next step from the database.
+The user is working with the canvas locally, but the state of the canvas is stored in the database.
+The main drawback of this approach is that the database will grow significantly over time.
+We have to store the entire state of the canvas for every step taken by the user.
+This could lead to performance issues in the future.
+
+Let's assume that there are 1000 users using the app.
+we have ~19 layers in the app.
+if the user places 10 shapes on each layer and modifies the properties of each shape 3 times, we have 570 actions taken by the user for each layer.
+If we store the entire state of the canvas for every action taken by the user, we have 10,830 actions stored in the database for each user.
+If we have 1000 users, we have 10,830,000 actions stored in the database for a single day. This number will grow drastically over time.
```

Comment:
Good point here! Since this question has no clear answer and is different from app to app, I've updated this part with the comparison showing the disadvantages of this approach